src/Abstractions/NexusMods.Abstractions.Settings/SettingsBuilder/IPropertyUIBuilder.cs

```diff
@@ -29,28 +41,34 @@ public interface IWithDescriptionStep
         /// <remarks>
         /// This property allows for Markdown.
         /// </remarks>
-        IOptionalStep WithDescription(string description);
+        IConfigureValueContainerStep WithDescription(string description);
+    }
+
+    public interface IConfigureValueContainerStep
+    {
+        IRequiresRestartStep UseBooleanContainer();
```

Comment:
Does this mean that this can be called on any value type? Even if it isn't bool?

Reply:
Sadly yes. You can't do `where TProperty : bool`, and the only way to have this check at compile time is by using an extension method:

```
public static class Extensions
{
    public static IPropertyUIBuilder<TSettings, bool>.IRequiresRestartStep UseBooleanContainer<TSettings>(
        this IPropertyUIBuilder<TSettings, bool>.IConfigureValueContainerStep step
    ) where TSettings : class, ISettings, new()
    {
        // ...
    }
}
```

However, this doesn't really work because the extension method has to have a body and has to return an actual value while the interface puts the method implementation requirement on the actual implementation, which isn't public.